sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -0,0 +1,206 @@
+    name?: string;
+}
+
+// @public
+export interface ContainerRegistryEventData {
+    action?: string;
+    actor?: ContainerRegistryEventActor;
+    id?: string;
+    request?: ContainerRegistryEventRequest;
+    source?: ContainerRegistryEventSource;
+    target?: ContainerRegistryEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryEventRequest {
+    addr?: string;
+    host?: string;
+    id?: string;
+    method?: string;
+    useragent?: string;
+}
+
+// @public
+export interface ContainerRegistryEventSource {
+    addr?: string;
+    instanceID?: string;
+}
+
+// @public
+export interface ContainerRegistryEventTarget {
+    digest?: string;
+    length?: number;
+    mediaType?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    url?: string;
+}
+
+// @public
+export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData & {};
+
+// @public
+export type ContainerRegistryImagePushedEventData = ContainerRegistryEventData & {};
+
+// @public
+export class EventGridClient {
+    constructor(endpointUrl: string, credential: KeyCredential | SignatureCredential, options?: EventGridClientOptions);
+    readonly apiVersion: string;
+    readonly endpointUrl: string;
+    generateSharedAccessSignature(expiresOnUtc: Date): string;
+    sendCloudEvents(events: CloudEvent<any>[], options?: SendCloudEventsOptions): Promise<RestResponse>;
+    sendCustomSchemaEvents(events: Record<string, any>[], options?: SendCustomSchemaEventsOptions): Promise<RestResponse>;
+    sendEvents(events: EventGridEvent<any>[], options?: SendEventsOptions): Promise<RestResponse>;
+}
+
+// @public
+export type EventGridClientOptions = PipelineOptions;
+
+// @public
+export class EventGridConsumer {
+    constructor(decoders?: Record<string, CustomEventDataDecoder>);
+    // Warning: (ae-forgotten-export) The symbol "CustomEventDataDecoder" needs to be exported by the entry point index.d.ts
+    //
+    // (undocumented)
+    readonly customDecoders: Record<string, CustomEventDataDecoder>;
+    decodeCloudEvents(encodedEvents: string): Promise<CloudEvent<unknown>[]>;
+    decodeCloudEvents(encodedEvents: object): Promise<CloudEvent<unknown>[]>;
+    decodeEventGridEvents(encodedEvents: string): Promise<EventGridEvent<unknown>[]>;
+    decodeEventGridEvents(encodedEvents: object): Promise<EventGridEvent<unknown>[]>;
+}
+
+// @public
+export interface EventGridEvent<T> {
+    data: T;
+    dataVersion: string;
+    eventTime: Date;
+    eventType: string;
+    id: string;
+    subject: string;
+    topic?: string;
+}
+
+// @public
+export class EventGridSharedAccessSignatureCredential implements SignatureCredential {
+    constructor(signature: string);
+    get signature(): string;
+    update(newSignature: string): void;
+}
+
+// @public
+export interface EventHubCaptureFileCreatedEventData {
+    eventCount?: number;
+    fileType?: string;
+    fileurl?: string;
```

Comment:
fileUrl?

Reply:
So this is interesting, these data types are generated from the definitions that the individual service teams publish into the rest specs repository (you can see the definition of this specific type [here](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-event-hubs)).  They are also documented inside the [event grid docs about system topics](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-event-hubs#event-grid-event-schema) and it turns out that these two sources disagree on the casing of the member!

This is, unfortunately, not the first time I have run into a difference between what is documented and what the service actually sends.

I'll dig into what the actual property name is the service is using and if it's `fileUrl`, I will update the Rest API specs and re-generate this type.

My goal here is to use the same names and casings as what the service team is sending.  We do limited translations with the mappers to do some stuff like "date strings" -> `Date` objects, but I want the shapes of these types to match the underlying JSON structure when possible.

Let me know if that doesn't sound reasonable.

Reply:
Generally speaking we are in the business of massaging service types into idiomatic types for the language, including adjusting name where necessary. Especially in cases where the casing is inconsistent internally we aught to just do the thing that users will expect (so, `fileUrl`, `userAgent`, etc.).

Reply:
> Generally speaking we are in the business of massaging service types into idiomatic types for the language, including adjusting name where necessary. Especially in cases where the casing is inconsistent internally we aught to just do the thing that users will expect (so, fileUrl, userAgent, etc.).

Okay, sounds good.  After thinking about this more, I'm comfortable with this model as well.